File: majorProject/src/main/java/com/aocc/majorproject/ui/ComposeOverlayBridge.java
package com.aocc.majorproject.ui;

import com.aocc.majorproject.Player;

/**
 * Java-facing bridge to the Jetpack Compose menu overlay hosted by {@code AndroidGame}.
 */
public interface ComposeOverlayBridge {

    interface MainMenuListener {
        void onPlay();

        void onTutorial();

        void onSignIn();

        void onLeaderboards();

        void onAchievements();

        boolean isLoggedIn();
    }

    interface SettingsListener {
        void onResumeGame();

        void onMenu();

        void onToggleSound();

        void onToggleMusic();

        void onFlatTilt();

        void onTiltedTilt();

        void onCustomTilt();

        boolean isSoundOn();

        boolean isMusicOn();

        int getTiltMode();
    }

    void showMainMenu(MainMenuListener listener);

    void showSettings(String prompt, boolean showMenuButton, SettingsListener listener);

    void hide();

    void refreshSettings();
}

File: majorProject/src/main/kotlin/com/aocc/majorproject/ui/ComposeOverlayBridge.kt
package com.aocc.majorproject.ui

import com.aocc.majorproject.Player

/**
 * Java-facing bridge to the Jetpack Compose menu overlay hosted by `AndroidGame`.
 */
interface ComposeOverlayBridge {

    interface MainMenuListener {
        fun onPlay()
        fun onTutorial()
        fun onSignIn()
        fun onLeaderboards()
        fun onAchievements()
        fun isLoggedIn(): Boolean
    }

    interface SettingsListener {
        fun onResumeGame()
        fun onMenu()
        fun onToggleSound()
        fun onToggleMusic()
        fun onFlatTilt()
        fun onTiltedTilt()
        fun onCustomTilt()
        fun isSoundOn(): Boolean
        fun isMusicOn(): Boolean
        fun getTiltMode(): Int
    }

    fun showMainMenu(listener: MainMenuListener)
    fun showSettings(prompt: String, showMenuButton: Boolean, listener: SettingsListener)
    fun hide()
    fun refreshSettings()
}
